src/main/scripts/src/app/core/rest/search.service.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *
+ * Copyright (C) since 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+
+import {Injectable} from '@angular/core';
+import {Http} from '@angular/http';
+import {Observable} from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+
+import {WorkspaceService} from '../workspace.service';
+import {Suggestion} from '../dto/suggestion';
+
+@Injectable()
+export class SearchService {
+
+  constructor(private http: Http, private workspaceService: WorkspaceService) {
+  }
+
+  public suggest(text: string, type: string): Observable<Suggestion[]> {
+    return this.http
+      .get(`${this.searchPath()}/suggestion?text=${text}&type=${type}`)
+      .map(response => response.json() as Suggestion[]);
```

Comment:
Thanks, I haven't even used our `HttpClient` here. I will refactor this using a new Angular client.